coderd/coderdtest/coderdtest.go

```diff
@@ -306,6 +308,20 @@ func NewOptions(t testing.TB, options *Options) (func(http.Handler), context.Can
 	hangDetector.Start()
 	t.Cleanup(hangDetector.Close)
 
+	if options.WorkspaceUsageTracker == nil {
+		// Workspace usage tracking must be triggered manually in tests.
+		// To do this, pass in your own WorkspaceUsageTracker.
+		wutFlush := make(chan int)
+		wutTick := make(chan time.Time)
+		options.WorkspaceUsageTracker = workspaceusage.New(
+			options.Database,
+			workspaceusage.WithLogger(options.Logger.Named("workspace_usage_tracker")),
+			workspaceusage.WithFlushChannel(wutFlush),
+			workspaceusage.WithTickChannel(wutTick),
+		)
+	}
+	t.Cleanup(options.WorkspaceUsageTracker.Close)
```

Comment:
self-review: to avoid unintentional flakes, this is disabled by default in `coderdtest`. Any behaviour that depends on last_used_at being updated needs to create and pass in an instance of `workspaceusage.Tracker` and tick/flush manually. See `portforward_test.go` for what this looks like in practice.

Reply:
This is fair :+1:. Unfortunately will likely be a debug session for someone in the future. I cannot think of a way to detect if someone will want this in their test.